modules/dbparser/stateful-parser.c

```diff
@@ -0,0 +1,54 @@
+#include "stateful-parser.h"
+#include <string.h>
+
+
+void
+stateful_parser_set_inject_mode(StatefulParser *self, const gchar *inject_mode)
+{
+  self->inject_mode = stateful_parser_lookup_inject_mode(inject_mode);
```

Comment:
fixed in 33af212f47ad6949f60c3b1cf5821e5a69babe6c by pushing the validation up to the parser.